Microsoft.Toolkit.Uwp.UI.Animations/ConnectedAnimations/Connected.cs

```diff
@@ -0,0 +1,326 @@
+                }
+
+                if (e.NewValue is UIElement newAnchor)
+                {
+                    helper?.AttachElementToAnimatingElement(element, newAnchor);
+                }
+            });
+        }
+
+        private static void OnListItemKeyChanged(DependencyObject d, DependencyPropertyChangedEventArgs e)
+        {
+            if (!ApiInformationHelper.IsCreatorsUpdateOrAbove)
+            {
+                return;
+            }
+
+            var element = d as Windows.UI.Xaml.Controls.ListViewBase;
+            if (element == null)
+            {
+                return;
+            }
+
+            GetParentFrameAndExecuteAction(element, (frame) =>
+            {
+                var helper = GetConnectedAnimationHelper(frame);
+
+                if (e.OldValue is string oldKey)
+                {
+                    helper?.RemoveListItem(element, oldKey);
+                }
+
+                AddListViewBaseItemAnimationDetails(helper, element);
+            });
+        }
+
+        private static void OnListItemElementNameChanged(DependencyObject d, DependencyPropertyChangedEventArgs e)
+        {
+            if (!ApiInformationHelper.IsCreatorsUpdateOrAbove)
+            {
+                return;
+            }
+
+            var element = d as Windows.UI.Xaml.Controls.ListViewBase;
+            if (element == null)
+            {
+                return;
+            }
+
+            GetParentFrameAndExecuteAction(element, (frame) =>
+            {
+                var helper = GetConnectedAnimationHelper(frame);
+
+                if (e.OldValue is string oldElementName)
+                {
+                    var elementKey = GetListItemKey(element);
+                    if (elementKey != null)
+                    {
+                        helper?.RemoveListItem(element, elementKey);
+                    }
+                }
+
+                AddListViewBaseItemAnimationDetails(helper, element);
+            });
+        }
+
+        private static void GetParentFrameAndExecuteAction(FrameworkElement element, Action<Frame> action)
+        {
+            // get parent Frame
+            var frame = element.FindAscendant<Frame>();
+
+            if (frame == null)
+            {
+                RoutedEventHandler handler = null;
+                handler = (s, args) =>
+                {
+                    element.Loaded -= handler;
+                    frame = element.FindAscendant<Frame>();
+                    if (frame != null)
+                    {
+                        action(frame);
+                    }
+                };
+
+                element.Loaded += handler;
+            }
+            else
+            {
+                action(frame);
+            }
+        }
+
+        private static void AddListViewBaseItemAnimationDetails(ConnectedAnimationHelper helper, Windows.UI.Xaml.Controls.ListViewBase listViewBase)
+        {
```

Comment:
Check for helper being null and at the top and early exit. No need to do all of this work if the helper is null

Reply:
Done